src/utility.js

```diff
@@ -13,4 +13,43 @@ function isNumber(n) {
 
 function getConfigValue(config, key, defaultValue) {
 	return config.hasOwnProperty(key) ? config[key] : defaultValue;
+}
+
+function getDomain (url) {
+	var regex = /^(?:(?:https?:)?\/\/)?(?:www\.)?(.*?)\//;
+	return url.match(regex)[1];
+}
+
+function getClickThroughURL (ad) {
+	var creative =  ad._ad.creatives[0];
+	return creative
+		? creative.clickThroughUrl
+		: '';
+}
+
+function exitFullscreenImpl() {
+	if(document.exitFullscreen) {
+		document.exitFullscreen()
+	} else if(document.webkitExitFullscreen) {
+		document.webkitExitFullscreen()
+	} else if(document.mozCancelFullScreen) {
+		document.mozCancelFullScreen()
+	} else if(document.msExitFullscreen) {
+		document.msExitFullscreen()
+	}
+}
+
+function getAgregatedAdTime (adBreak) {
+	return adBreak._playableAdsRemaining
+		.reduce(function(aggregatedTime, ad) {
+			try {
+				return getCreativeDuration(ad) + aggregatedTime;
+			} catch (e) {
+				return aggregatedTime;
+			}
+	    }, 0)
+}
+
+function getCreativeDuration (ad) {
+	return ad.creatives[0].duration;
```

Comment:
Can it ever happen that ad.creatives[0] is `null` or `undefined`?

Reply:
we always have creative, if the ad is getting played.